src/algo_tx.h

```diff
@@ -54,19 +68,58 @@ struct txn_asset_config {
   struct asset_params params;
 };
 
+struct state_schema {
+  uint64_t num_uint;
+  uint64_t num_byteslice;
+};
+
+#define MAX_ACCT 2
+typedef uint8_t accounts_t[MAX_ACCT][32];
+
+#define MAX_ARG 2
+#define MAX_ARGLEN 32
+#define MAX_FOREIGN_APPS 1
+#define MAX_APPROV_LEN 128
+#define MAX_CLEAR_LEN 32
+
+struct txn_application {
+  uint64_t id;
+  uint64_t oncompletion;
+
+  uint8_t accounts[MAX_ACCT][32];
+  size_t num_accounts;
+
+  uint64_t foreign_apps[MAX_FOREIGN_APPS];
+  size_t num_foreign_apps;
+
+  uint8_t app_args[MAX_ARG][MAX_ARGLEN];
+  size_t app_args_len[MAX_ARG];
```

Comment:
What do you think of defining a `struct app_arg { uint8_t arg[MAX_ARGLEN]; size_t arg_len; };` and then defining this to be `struct app_arg app_args[MAX_ARG];`?  I think this might slightly simplify some code in encode/decode where you just need to pass an individual `struct app_arg*` to decode/encode, instead of passing matching buffers and lengths.